da/celestia/celestia.go

```diff
@@ -111,60 +127,101 @@ func (c *DataAvailabilityLayerClient) SubmitBlock(ctx context.Context, block *ty
 }
 
 // CheckBlockAvailability queries DA layer to check data availability of block at given height.
-func (c *DataAvailabilityLayerClient) CheckBlockAvailability(
-	ctx context.Context, dataLayerHeight uint64,
-) da.ResultCheckBlock {
-	shares, err := c.client.NamespacedShares(ctx, c.namespaceID, dataLayerHeight)
-	code := dataRequestErrorToStatus(err)
-	if code != da.StatusSuccess {
+func (c *DataAvailabilityLayerClient) CheckBlockAvailability(ctx context.Context, dataLayerHeight uint64) da.ResultCheckBlock {
+	header, err := c.rpc.Header.GetByHeight(ctx, dataLayerHeight)
+	if err != nil {
+		return da.ResultCheckBlock{
+			BaseResult: da.BaseResult{
+				Code:    da.StatusError,
+				Message: err.Error(),
+			},
+		}
+	}
+	if header.DAH == nil {
+		return da.ResultCheckBlock{
+			BaseResult: da.BaseResult{
+				Code:     da.StatusSuccess,
+				DAHeight: dataLayerHeight,
+			},
+			DataAvailable: false,
+		}
+	}
+	err = c.rpc.Share.SharesAvailable(ctx, header.DAH)
+	if err != nil {
+		if err.Error() == ErrNotAvailable.Error() {
```

Comment:
I'd recommend `error.Is(err, ErrNotAvailable)` if `SharesAvailable` doesn't extend `ErrNotAvailable` or if it potentially does it would do `strings.Contains(err.Error(), ErrNotAvailable.Error())`

Reply:
The error would need to be imported so `error.Is` does't work.

Reply:
Using `strings.Contains` for now as the error type can't be imported without a dependency.